pyphare/pyphare_tests/test_pharesee/test_geometry.py

```diff
@@ -0,0 +1,528 @@
+                               cell_width=cell_width,
+                               refinement_ratio=refinement_ratio,
+                               refinement_boxes=refinement_boxes)
+
+
+
+
+    def test_overlaps(self):
+        expected = {0 :
+                        # Middle overlap, for all quantities
+                        [  {"box":Box(28, 38),'offset':(0,0)},
+
+                        {"box": Box(28, 37),"offset":(0,0)},
+                        {"box": Box(28, 37),"offset":(0,0)},
+                        {"box": Box(28, 37),'offset': (0,0)},
+                        {"box": Box(28, 38),"offset":(0,0)},
+                        {"box": Box(28, 38),"offset":(0,0)},
+                        {"box": Box(32,33),"offset":(0,0)},
+
+                        # left side overlap with periodicity, for all quantities
+                          {"box": Box(-5, 5),"offset":(0,-65)},
+                        # right side overlap with periodicity, for all quantities
+                        {"box": Box(60, 70),"offset": (65,0)},
+                        {"box": Box(-5, 4),"offset":(0,-65)},
+                        {"box": Box(60, 69),"offset": (65,0 )},
+                        {"box": Box(-5, 4),"offset":(0,-65)},
+                        {"box": Box(60, 69),"offset": (65,0)},
+                        {"box": Box(-5, 4),"offset":(0,-65)},
+                        {"box": Box(60, 69),"offset":(65, 0)},
+                        {"box": Box(-5, 5),"offset":(0,-65)},
+                        {"box": Box(60, 70),"offset":(65, 0)},
+                        {"box": Box(-5, 5),"offset":(0,-65)},
+                        {"box": Box(60, 70),"offset":(65, 0)},
+                        {"box": Box(-1,0),"offset":(0,-65)},
+                        {"box": Box(64, 65),"offset":(65 ,0)}
+                    ],
+
+            1:  # level 1
+                [
+                    {"box": Box(59, 65),'offset': (0, 0)},
+                    {"box": Box(59, 64),"offset": (0, 0)},
+                    {"box": Box(59, 64),"offset": (0, 0)},
+                    {"box": Box(59, 64),'offset': (0, 0)},
+                    {"box": Box(59, 65),"offset": (0, 0)},
+                    {"box": Box(59, 65),"offset": (0, 0)}
+                ]
+
+
+        }
+
+
+        overlaps = hierarchy_overlaps(self.hierarchy)
+
+        for ilvl, lvl in enumerate(self.hierarchy.patch_levels):
+            self.assertEqual(len(expected[ilvl]), len(overlaps[ilvl]))
+
+            for exp, actual in zip(expected[ilvl], overlaps[ilvl]):
+
+                act_box    = actual["box"]
+                act_offset = actual["offset"]
+                exp_box    = exp["box"]
+                exp_offset = exp["offset"]
+
+                self.assertEqual(act_box, exp_box)
+                self.assertEqual(act_offset, exp_offset)
+
+
+
+
+    def test_touch_border(self):
+
+        self.assertFalse(touch_domain_border(Box(10,20), self.hierarchy.domain_box, "upper"))
```

Comment:
to see if it touches the border, we still have 5 ghost cells, and not a number depending on the onterp_order ?